src/renderer/StoreChangeObserver.tsx

```diff
@@ -188,6 +190,21 @@ const StoreChangeObserver = () => {
     ipc.setConfidenceLinesEnabled(words !== null);
   }, [isShowingConfidenceUnderlines, words]);
 
+  // Autosave after currentProject changes.
+  useEffect(() => {
+    if (
+      currentProject &&
+      currentProject.transcription &&
+      currentProject.projectFilePath
+    ) {
+      if (collab !== null && collab.isHost) return; // Disallow autosave if you are not the host in collab
+
+      ipc.saveProject(currentProject);
+
+      currentProject.isEdited = false; // set the dirtiness to clean!
```

Comment:
You need to dispatch this, if you just edit it it won't update the store (or at the very least, if it does work, it shouldn't and will cause unexpected side effects since redux is meant to be immutable)

You can do this by dispatching a `projectSaved` action. Just be aware this will cause an infinite loop so check that `currentProject.isEdited` is `true` here, otherwise early exit

Reply:
Will do

Reply:
Just realised that I was using `saveProject` from ipc, which kind of requires repeated code, at least when comparing to `saveProject` from `renderer/file`, im not sure how jank it is to import that function instead of rewriting some logic to dispatch a projectSaved